modules/apm/src/main/java/org/elasticsearch/tracing/apm/APMTracer.java

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.tracing.apm;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.util.automaton.Automata;
+import org.apache.lucene.util.automaton.Automaton;
+import org.apache.lucene.util.automaton.CharacterRunAutomaton;
+import org.apache.lucene.util.automaton.Operations;
+import org.apache.lucene.util.automaton.RegExp;
+import org.elasticsearch.Version;
+import org.elasticsearch.common.component.AbstractLifecycleComponent;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.Maps;
+import org.elasticsearch.common.util.concurrent.ConcurrentCollections;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.core.Releasable;
+import org.elasticsearch.tasks.Task;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.tracing.apm.APMAgentSettings.APM_ENABLED_SETTING;
+import static org.elasticsearch.tracing.apm.APMAgentSettings.APM_TRACING_NAMES_EXCLUDE_SETTING;
+import static org.elasticsearch.tracing.apm.APMAgentSettings.APM_TRACING_NAMES_INCLUDE_SETTING;
+
+/**
+ * This is an implementation of the {@link org.elasticsearch.tracing.Tracer} interface, which uses
+ * the OpenTelemetry API to capture spans.
+ * <p>
+ * This module doesn't provide an implementation of the OTel API. Normally that would mean that the
+ * API's default, no-op implementation would be used. However, when the APM Java is attached, it
+ * intercepts the {@link GlobalOpenTelemetry} class and provides its own implementation instead.
+ */
+public class APMTracer extends AbstractLifecycleComponent implements org.elasticsearch.tracing.Tracer {
+
+    private static final Logger LOGGER = LogManager.getLogger(APMTracer.class);
```

Comment:
I think we started using lowercase `logger` throughout at some point didn't we?

Reply:
I didn't think so, but a quite check showed me that lower-case is way more popular, so I've changed it.